### ISSUE TITLE
DCC errors out looking for bam index/sorted bam

### ISSUE THREAD

**Issue by ShobanaSekar** [dieterich-lab/DCC#44]
Hi,

I am running into the attached error when running DCC where it says it cannot find the bam index or that the bam is not sorted by coordinate, but both conditions are true for my dataset.

[DCC_ADCC_ErrorLog.txt](https://github.com/dieterich-lab/DCC/files/1493218/DCC_ADCC_ErrorLog.txt)

This is my command line:
DCC @data/samplesheet -mt1 @data/mate1 -mt2 @data/mate2 -D -N -R ${Repeats} -an ${Annotation} -Pi -F -M -Nr 2 1 -fg -G -A ${Reference}

This is the content of _tmp_DCC folder:

[tmp_DCC_ADCClog.txt](https://github.com/dieterich-lab/DCC/files/1493229/tmp_DCC_ADCClog.txt)

Appreciate your help!

Best,
Shobana

**Comment by tjakobi** [comment]
Dear Shobana,

please try to use the -B flag with a file containing the BAM files as suggested by the error message:  

```
  -B BAM [BAM ...], --bam BAM [BAM ...]
                        A file specifying the mapped BAM files from which host
                        gene expression is computed; must have the same order
                        as input chimeric junction files
```
Your input files are not following the standard STAR naming scheme which is why the automatic guessing step fails.

Please let me know if that solves your problem.

Cheers,
Tobias

**Comment by ShobanaSekar** [comment]
Thank you Tobias. I tried that as well but I still end up with the same error.

This is my command line:
DCC data/samplesheet -mt1 data/mate1 -mt2 data/mate2 -B data/bamfilelist -D -N -R ${Repeats} -an ${Annotation} -Pi -F -M -Nr 2 1 -fg -G -A ${Reference}

I added the bam file list (transcriptome out bam from STAR).
This is my bam file list (same order as chimeric junction files)
[bamfilelist.txt](https://github.com/dieterich-lab/DCC/files/1500192/bamfilelist.txt)

And here are the logs:
[DCC-2017-11-23_12-44.log.txt](https://github.com/dieterich-lab/DCC/files/1500193/DCC-2017-11-23_12-44.log.txt)

Thanks again for your help!

-Shobana

**Comment by tjakobi** [comment]
Dear Shobana,

could you please verify your version of pysam?

```
pip list | grep sam
pysam (0.10.0)
```
Cloud you additionally list the output of
```
la -la /scratch/ssekar/ADCC_circRNA_rerun/downsampled/DCC/STAR/STAR_OUT/ADCC_0001_PC_L02652.sub.proj/Joint/ADCC_0001_PC_L02652.sub.*
```
I'd like to see how the BAM files are their indices are named.

Cheers,
Tobias

**Comment by ShobanaSekar** [comment]
My pysam version is:
pysam (0.10.0)

Here is the listing of the Joint directory for this sample:
[Sample_0001_JointFolder_listing.txt](https://github.com/dieterich-lab/DCC/files/1504053/Sample_0001_JointFolder_listing.txt)


Thanks,
Shobana

**Comment by tjakobi** [comment]
Dear Shobana,

sorry for the delay. I'm  having a hard time reproducing your error.
Just to make sure we're on the same page:

* Make sure you're running the latest version of DCC:
```
$ git clone https://github.com/dieterich-lab/DCC.git
$ cd DCC
$ python setup.py install --user --force
```
I have the impression the error may already haven been fixed in a git commit after the 0.4.4 release. 

Cheers,
Tobias

**Comment by ShobanaSekar** [comment]
Hi Tobias,

I apologize for the delay. I initially had some installation issues (permissions related) so my hpc admin installed it for me. After your reply I tried re-installing in my local home with the --user option and it went through fine. Both were 0.4.4 version but am not sure why the first installation is not running but the second one is. But thanks a lot for your help and appreciate your patience!

Best,
Shobana
